### PR TITLE
[PM-5714] Added docs for MAUI mobile app

### PR DESCRIPTION
**Pull request by fedemkr**
## Objective

Add docs for MAUI mobile app:
- Update requirements
- Replace `Xamarin` words with `MAUI` where appropriate
- Add `CustomConstants` section and how to use them (F-Droid and Unit tests)
- How to build/run unit tests
- How to build Android/iOS from the CLI given that Visual Studio may not work on some circumstances
- Added AndroidX Credentials section in Android to explain why the local nuget package is needed for this library and where to find it.

Added custom words `F-Droid` and `precompiler` to be used in the docs.

### CODE REVIEW

File: custom-words.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ clickjacking
 CODEOWNERS
 CQRS
 dockerized
+FDROID
```
Comment (withinfocus) [inline custom-words.txt:11]: ⛏️ This is technically "F-Droid".

File: docs/architecture/mobile-clients/overview.md [hunk 2]
```diff
@@ -16,5 +16,5 @@ differs:
 
 ## Presentation
 
-The presentation layer is implemented using `Xamarin.Forms` for the mobile apps, except for the
-watchOS one which uses `SwiftUI` [see ADR](../adr/watchOS-use-swift)
+The presentation layer is implemented using `.NET MAUI` for the mobile apps, except for the watchOS
```
Comment (withinfocus) [inline docs/architecture/mobile-clients/overview.md:19]: ⛏️ "MAUI" or ".NET MAUI" aren't code references in my opinion and we shouldn't put backticks all over the place.

File: docs/getting-started/clients/mobile/android/index.md [hunk 3]
```diff
@@ -5,13 +5,16 @@
...
-1.  Visual Studio 2022
-2.  [.NET 7 (latest)](https://dotnet.microsoft.com/en-us/download/dotnet/7.0)
+1.  Visual Studio 2022 / VS Code
+2.  [.NET 8 (latest)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
     - Note: Even if you have an M1 Mac with ARM 64 architecture, you can install all x64 SDKs to run
       android
-3.  [Xamarin (Android)](https://learn.microsoft.com/en-us/xamarin/get-started/installation/?pivots=macos-vs2022)
-4.  Android SDK 33
-    - You can use the SDK manager in [Xamarin, Visual Studio][xamarin-vs], or [Android
+    - On Visual Studio for Mac you may need to turn on the feature for NET 8 going to Visual
```
Comment (withinfocus) [inline docs/getting-started/clients/mobile/android/index.md:12]:
```suggestion
    - On Visual Studio for Mac you may need to turn on the feature for .NET 8 by going to Visual
```

File: docs/getting-started/clients/mobile/index.md [hunk 4]
```diff
@@ -46,3 +46,49 @@ available to internal team members. We do not have any documentation for communi
 this time.
 
 </community>
+
+## Unit tests
+
+TL;DR; In order to run unit tests add the argument `/p:CustomConstants=UT` on the dotnet command for
+building/running.
```
Comment (withinfocus) [inline docs/getting-started/clients/mobile/index.md:53]: 🎨 You could surround this with a note or info block to make it more prominent.

File: docs/getting-started/clients/mobile/index.md [hunk 5]
```diff
@@ -46,3 +46,49 @@ available to internal team members. We do not have any documentation for communi
 this time.
 
 </community>
+
+## Unit tests
+
+TL;DR; In order to run unit tests add the argument `/p:CustomConstants=UT` on the dotnet command for
```
Comment (withinfocus) [inline docs/getting-started/clients/mobile/index.md:52]:
```suggestion
TL;DR; In order to run unit tests add the argument `/p:CustomConstants=UT` on the `dotnet` command for
```

File: docs/getting-started/clients/mobile/ios/index.mdx [hunk 6]
```diff
@@ -10,9 +10,15 @@ import TabItem from "@theme/TabItem";
 
 ## Requirements
 
-- [.NET 7 (latest)](https://dotnet.microsoft.com/en-us/download/dotnet/7.0)
-- Xamarin (iOS)
-- A Mac with Xcode 15 installed
+1.  Visual Studio 2022 / VS Code
+2.  [.NET 8 (latest)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
+    - Note: Even if you have an M1 Mac with ARM 64 architecture, you can install all x64 SDKs to run
+      android
```
Comment (withinfocus) [inline docs/getting-started/clients/mobile/ios/index.mdx:16]:
```suggestion
      Android
```

**Review by withinfocus — CHANGES_REQUESTED** [review]
Few small quality items.

File: docs/architecture/mobile-clients/overview.md (hunk 2 above)
Comment (fedemkr) [reply docs/architecture/mobile-clients/overview.md]: @withinfocus Roger that, this also applies to `SwiftUI`, `XCode`, `CoreData`, `Xamarin.Forms`. At the moment I use that in order to make the string stand out. Would you like Bold/Italics more? Or simply put it in the raw way?

Comment (withinfocus) [reply docs/architecture/mobile-clients/overview.md]: I think undecorated is still best for uniform presentation.

**Review by withinfocus — DISMISSED** [review]
Not sure why the build is failing but this looks good to me.

**Comment by withinfocus** [comment]
Once you fix that link issue the mobile folks can provide final approval.

File: docs/getting-started/clients/mobile/ios/index.mdx [hunk 7]
```diff
@@ -247,12 +251,88 @@ Next, we need to configure your Visual Studio environment for development.
...
+`iossimulator-arm64`.
+
+:::tip Troubleshooting common mistake
+
+If you find the next error:
+
+> `error NETSDK1134`: Building a solution with a specific RuntimeIdentifier is not supported. If you
+> would like to publish for a single RID, specify the RID at the individual project level instead
+
+You almost surely are trying to build the app from the root folder. Instead go to _src/App_ and try
```
Comment (SaintPatrck) [inline docs/getting-started/clients/mobile/ios/index.mdx:289]: Project paths are surrounded by backticks in other setup guides.
```suggestion
You almost surely are trying to build the app from the root folder. Instead go to `src/App` and try
```

File: docs/getting-started/clients/mobile/watchos/index.mdx [hunk 8]
```diff
@@ -82,5 +82,5 @@ aforementioned.
 So one can debug one end (iOS) or the other (watchOS).
 
 If needed to check something on both ends "at the same time" (like to check why a message is not
-sent/arrived), one needs to use console logging or adapt part of the Xamarin code to the iOS stub
-app on Xcode and debug the synchronization from XCode.
+sent/arrived), one needs to use console logging or adapt part of the MAUI code to the iOS stub app
+on Xcode and debug the synchronization from XCode.
```
Comment (SaintPatrck) [inline docs/getting-started/clients/mobile/watchos/index.mdx:86]:
```suggestion
on Xcode and debug the synchronization from Xcode.
```

File: docs/getting-started/clients/mobile/ios/index.mdx [hunk 9]
```diff
@@ -247,12 +251,88 @@ Next, we need to configure your Visual Studio environment for development.
...
+
+```
+dotnet build -f net8.0-ios -c Debug -r ios-arm64
+```
+
+For simulator:
+
+```
+dotnet build -f net8.0-ios -c Debug -r iossimulator-x64
+```
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply docs/getting-started/clients/mobile/ios/index.mdx:276]: Thanks for the feedback Patrick 🎉 
This is because of VS for Mac that won't be updated anymore. VS Code should be able to build the projects correctly.
And I was referring to any kind of weird error presented by VS for Mac, like ditto errors; which are fixed by doing what's explained there but yes I can put it in a note container.

File: docs/getting-started/clients/mobile/index.md [hunk 10]
```diff
@@ -46,3 +46,56 @@ available to internal team members. We do not have any documentation for communi
...
+
+and to run the tests go to the same folder and run:
+
+```bash
+dotnet test -f net8.0 /p:CustomConstants=UT
+```
+
+Finally, when working on the `Core.Test` project or when wanting to use a Test runner, go to the
+`Directory.Build.props` (located in the root) and uncomment the line referencing `CustomConstants`
+so that everything is loaded accordingly in the project. Because of some issues the referenced
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply docs/getting-started/clients/mobile/index.md:85]: Thanks Andre, will do the changes 🎉  The only thing I think it's not needed is the last comma before the "and" so I'll omit that one.

**Review by SaintPatrck — DISMISSED** [review]

File: docs/architecture/mobile-clients/watchOS.md [hunk 11]
```diff
@@ -21,8 +21,8 @@ Then in the Extension we have a layered architecture:
 
 ## Integration with iOS
 
-The watchOS app is developed using `XCode` and `Swift` and we need to integrate it to the `Xamarin`
-iOS application.
+The watchOS app is developed using `XCode` and `Swift` and we need to integrate it to the `MAUI` iOS
```
Comment (Hinton) [inline docs/architecture/mobile-clients/watchOS.md:24]: We use code formatting for MAUI here, but leave it without in the overview.

File: docs/getting-started/clients/mobile/ios/index.mdx [hunk 12]
```diff
@@ -81,16 +85,16 @@ required.
...
 ![](./troubleshoot-generic-simulator.png)
 
 To work around this issue, try [downloading](https://developer.apple.com/download/all/) and
 installing an older version of Xcode from Apple (you can look for guidance on which Xcode version to
-use from the Xamarin.iOS [release notes](https://github.com/xamarin/xamarin-macios/releases). After
-installing the new version of Xcode, restart Visual Studio and load your project to verify your
-available simulator options.
+use from the Xamarin.iOS [release notes](https://github.com/xamarin/xamarin-macios/releases) (this
+applies to MAUI as well). After installing the new version of Xcode, restart Visual Studio and load
+your project to verify your available simulator options.
```
Comment (Hinton) [inline docs/getting-started/clients/mobile/ios/index.mdx:97]: Should we remove the old xamarin references here?

File: docs/architecture/mobile-clients/watchOS.md (hunk 11 above)
Comment (fedemkr) [reply docs/architecture/mobile-clients/watchOS.md:24]: I can remove the code formatting here as well

**Review by andrebispo5 — APPROVED** [review]
Looks good 🎉 
Just some merge conflicts to resolve.

File: docs/getting-started/mobile/index.md [hunk 13]
```diff
@@ -82,18 +82,18 @@ dotnet test -f net8.0 /p:CustomConstants=UT
 
 Finally, when working on the `Core.Test` project or when wanting to use a Test runner, go to the
 `Directory.Build.props` (located in the root) and uncomment the line referencing `CustomConstants`
-so that everything is loaded accordingly in the project. Because of some issues the referenced
-projects, e.g. `Core`, are only included when the `UT` constant is in place so by uncommenting the
-line the projects will be referenced and one can work on that project or run the tests from a Test
-runner.
+so that everything is loaded accordingly in the project. Because of some issues, the referenced
+projects, e.g. `Core`, are only included when the `UT` constant is in place. So, by uncommenting
```
Comment (withinfocus) [inline docs/getting-started/mobile/index.md:86]:
```suggestion
projects, e.g. `Core`, are only included when the `UT` constant is in place. By uncommenting
```

File: docs/getting-started/mobile/index.md [hunk 14]
```diff
@@ -82,18 +82,18 @@ dotnet test -f net8.0 /p:CustomConstants=UT
...
+this line, the projects will be referenced and one can work on that project or run the tests from a
```
Comment (withinfocus) [inline docs/getting-started/mobile/index.md:87]:
```suggestion
this line the projects will be referenced and one can work on that project or run the tests from a
```

File: docs/getting-started/mobile/ios/index.mdx [hunk 15]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
 
 ```
+cd src/App
 dotnet build -f net8.0-ios -c Debug -r iossimulator-x64
 ```
 
-If you find any errors regarding argon2Id library when building for `iossimulator-x64` then try to
-build for device `ios-arm64` as there might be some issues regarding arch differences and
-requirements for the argon2Id library to be used in `iossimulator-x64`.
+You can also use the IDE but keep in mind the next tips:
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:278]:
```suggestion
You can also use the IDE but keep in mind:
```

File: docs/getting-started/mobile/ios/index.mdx [hunk 16]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
+
+:::tip Argon2Id
+
+If you find any errors regarding argon2Id library when building for simulator, please be sure that
+you are building for runtime identifier `iossimulator-x64` as currently the library doesn't support
+`iossimulator-arm64`.
+
+:::
+
+:::tip Troubleshooting common mistake
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:296]:
```suggestion
:::tip Troubleshooting common mistakes
```

File: docs/getting-started/mobile/ios/index.mdx [hunk 17]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
+
+If you find the next error:
+
+> `error NETSDK1134`: Building a solution with a specific RuntimeIdentifier is not supported. If you
+> would like to publish for a single RID, specify the RID at the individual project level instead
+
+You almost surely are trying to build the app from the root folder. Instead go to `src/App` and try
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:303]:
```suggestion
you almost surely are trying to build the app from the root folder. Instead go to `src/App` and try
```

File: docs/getting-started/mobile/ios/index.mdx [hunk 18]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
+building again.
+
+:::
+
+### Argon2Id library loading
+
+The Argon2Id library (`libargon2.a`) is loaded using `MTouchExtraArgs` in almost all csproj of the
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:310]: ⛏️ Might be cleaner to just say "projects" instead of "csproj".

File: docs/getting-started/mobile/ios/index.mdx [hunk 19]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
+solution. In order to make this simpler a property was added into **Directory.Build.props** called
+`Argon2IdLoadMtouchExtraArgs` which has the code to fill in the extra args parameter. Each csproj is
+configured with this property so this is only added on the correct runtime identifiers and we can
+build the app successfully on each case.
+
+### Ignoring extensions / watchOS app
+
+Sometimes we need to quickly build the app or maybe some configuration on the iOS extensions or the
+watchOS app gets in the way. So in order to have a fast way to only care about the main app two
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:319]:
```suggestion
watchOS app gets in the way. In order to have a fast way to only care about the main app two
```

File: docs/getting-started/mobile/ios/index.mdx [hunk 20]
```diff
@@ -259,22 +259,93 @@ Next, we need to configure your Visual Studio environment for development.
...
+properties were added to the **Directory.Build.Props** to help with this:
+
+- `IncludeBitwardeniOSExtensions`: If `True` then all the iOS extensions will be included on the
+  building of the main app, otherwise they will be skipped.
+- `IncludeBitwardenWatchOSApp`: If `True` then the watchOS app will be included on the building of
+  the main app, otherwise it will be skipped.
+
+:::warning Shared code
+
+Toggling these off can provide a faster developer experience which is really useful on a lot of
```
Comment (withinfocus) [inline docs/getting-started/mobile/ios/index.mdx:329]:
```suggestion
Toggling these off can provide a faster developer experience which is really useful in a lot of
```

**Review by withinfocus — CHANGES_REQUESTED** [review]
Few minor things just from the changes since I last reviewed.

**Review by withinfocus — APPROVED** [review]
